test/e2e/customization_test.go

```diff
@@ -0,0 +1,176 @@
+package e2e
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/testframework"
+)
+
+const (
+	testCustomProductName = "test-e2e-product-name"
+	testCustomLogoName    = "test-e2e-configmap"
+	testCustomLogoKey     = "pic.jpg"
+	testMountPath         = "/app/" + testCustomLogoKey
+)
+
+// Test prep - setup the client used by each test
+func setupCustomTestCase(t *testing.T) (*testframework.Clientset, operatorsv1.ConsoleCustomization) {
+	client := testframework.MustNewClientset(t, nil)
+	// Get the original operator config
+	originalConfig, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("could not get operator config, %v", err)
+	}
+	originalConfigCustomization := originalConfig.Spec.Customization
+
+	return client, originalConfigCustomization
+}
+
+func cleanupCustomizationTestCase(t *testing.T, client *testframework.Clientset, originalConfigCustomization operatorsv1.ConsoleCustomization) {
+
+	setOperatorConfigCustomization(t, client, originalConfigCustomization)
+
+	err := client.ConfigMaps("openshift-config").Delete(testCustomLogoName, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("could not delete test configmap, %v", err)
+	}
+
+}
+
+// TestOperatorConfigCustomization() tests that changing the customization values on the operator-config
+// will result in the customization being set on the console-config in openshift-console.
+// Implicitly it ensures that the operator-config customization overrides customization set on
+// console-config in openshift-config-managed, if the managed configmap exists.
+func TestOperatorConfigCustomization(t *testing.T) {
```

Comment:
Contemplating if we need to check that the console pods are happy, or if they crashloop, since we can't easily open a browser & verify the logo appeared.

This was likely a good question for the previous `providers_test.go` as well, but we didn't think about it.  I would expect if we did want to do this as a basic "it didn't fall over" check, it could be a write-once-use-everywhere shared func.

Reply:
Yes, we should test not only the console pods but also the console-operator pods since I noticed the e2e will run and all fail when the pods are having an issue (which took me a second to realize what was going on).